src/http_server.cc

```diff
@@ -336,29 +336,27 @@ JsonBytesArrayByteSize(
     triton::common::TritonJson::Value& tensor_data, size_t* byte_size)
 {
   *byte_size = 0;
-
-  for (size_t i = 0; i < tensor_data.ArraySize(); i++) {
-    triton::common::TritonJson::Value el;
-    RETURN_IF_ERR(tensor_data.At(i, &el));
-
-    // Recurse if not last dimension...
-    TRITONSERVER_Error* assert_err =
-        el.AssertType(triton::common::TritonJson::ValueType::ARRAY);
-    if (assert_err == nullptr) {
+  // Recurse if not last dimension...
+  TRITONSERVER_Error* assert_err =
+      tensor_data.AssertType(triton::common::TritonJson::ValueType::ARRAY);
+  if (assert_err == nullptr) {
+    for (size_t i = 0; i < tensor_data.ArraySize(); i++) {
+      triton::common::TritonJson::Value el;
+      RETURN_IF_ERR(tensor_data.At(i, &el));
       size_t byte_size_;
       RETURN_IF_ERR(JsonBytesArrayByteSize(el, &byte_size_));
       *byte_size += byte_size_;
-    } else {
-      // Serialized data size is the length of the string itself plus
-      // 4 bytes to record the string length.
-      const char* str;
-      size_t len = 0;
-      RETURN_MSG_IF_ERR(
-          el.AsString(&str, &len), "Unable to parse JSON bytes array");
-      *byte_size += len + sizeof(uint32_t);
     }
-    TRITONSERVER_ErrorDelete(assert_err);
+  } else {
+    // Serialized data size is the length of the string itself plus
+    // 4 bytes to record the string length.
+    const char* str;
+    size_t len = 0;
+    RETURN_MSG_IF_ERR(
+        tensor_data.AsString(&str, &len), "Unable to parse JSON bytes array");
```

Comment:
FYI I think this macro leaks memory on errors: 
https://github.com/triton-inference-server/server/blob/f4d427841767601134ad7d13f1952fe7cbb34f72/src/common.h#L65-L69

It looks like it creates a new error and returns early, not deleting the original error.